llama-index-core/llama_index/core/workflow/utils.py

```diff
@@ -54,12 +55,26 @@ def validate_step_signature(fn: Callable) -> None:
         raise WorkflowValidationError(msg)
 
 
-def get_steps_from_class(_class: object) -> dict:
+def get_steps_from_class(_class: object) -> Dict[str, Callable]:
```

Comment:
This is a new function expecting a class, it's used to check for duplicate step names in `add_step`. Note that the predicate we pass to `getmembers` is different, as a class doesn't have "methods" until is instantiated.